File: lecture_002(Normal)/l_013_pythagoreantriplets.cpp
#include<iostream>
using namespace std;

void pythagoreantriplets(int a, int b, int c) {
    int hypotenuse = a;
    if(b>=hypotenuse) {
        hypotenuse = b;
    }
    if(c>=hypotenuse) {
        hypotenuse = c;
    }
    bool flag = false;
    if(hypotenuse == a) {
        if((b * b + c * c) == (a * a)) {
        	flag = true;
        }
        cout<<flag;
    } 
    else if(hypotenuse == b) {
        if((a * a + c * c) == (b * b)) {
            flag = true;
        }
        cout<<flag;
    } 
    else {
        if((a * a + b * b) == (c * c)) {
            flag = true;
        }
        cout<<flag;
    }
}
int main(int args, char**argv) {
    int a,b,c;
    cin>>a>>b>>c;
    pythagoreantriplets(a,b,c);
    
    return 0;
}

File: lecture_002(Normal)/l_013_pythagoreantriplets.java
import java.util.Scanner;
public class l_013_pythagoreantriplets {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int a = scn.nextInt();
        int b = scn.nextInt();
        int c = scn.nextInt();
        pythagoreantriplets(a,b,c);
    }
    public static void pythagoreantriplets(int a, int b, int c) {
        int hypotenuse;
        if(a>b && a>c) {
            hypotenuse = a;
        }
        else if(b>a && b>c) {
            hypotenuse = b;
        }
        else {
            hypotenuse = c;
        }
        if(hypotenuse == a) {
            boolean flag = ((b * b + c * c) == (a * a));
            System.out.println(flag);
        } 
        else if(hypotenuse == b) {
            boolean flag = ((a * a + c * c) == (b * b));
            System.out.println(flag);
        } 
        else {
            boolean flag = ((a * a + b * b) == (c * c));
            System.out.println(flag);
        }
    }
}